internal/provider/resource_managed_identity_access_rules.go

```diff
@@ -246,6 +258,7 @@ func (t *managedIdentityAccessRuleResource) Create(ctx context.Context,
 		AllowedServiceAccounts:    allowedServiceAccountsInput,
 		AllowedTeams:              allowedTeamsInput,
 		ModuleAttestationPolicies: policies,
+		VerifyStateLineage:        ptr.Bool(accessRule.VerifyStateLineage.ValueBool()),
```

Comment:
This value is optional so it should only be passed in to the create if it's provided.